### PR TITLE
Extend Outline Button test coverage with test for disabled state

### PR DESCRIPTION
**Pull request by clocksmith**
1st smaller PR sourced from larger PR: https://github.com/flutter/flutter/pull/18929

### CODE REVIEW

**Review by willlarche — APPROVED** [review]
LGTM! @HansMuller ?

File: packages/flutter/test/material/outline_button_test.dart [hunk 1]
```diff
@@ -44,78 +44,98 @@ void main() {
...
+    expect(tester.widget<OutlineButton>(disabledOutlineButton).enabled, false);
+
     expect(
-      outlineButton,
+      disabledOutlineButton,
+      paints
+        ..clipPath(pathMatcher: coversSameAreaAs(clipPath, areaToCompare: clipRect.inflate(10.0)))
+        ..path(color: disabledBorderColor, strokeWidth: borderWidth));
+
+    // Pump a new widget with no onPressed to make it enabled.
```
Comment (HansMuller) [inline packages/flutter/test/material/outline_button_test.dart]: ... with an onPressed callback, to make it enabled

File: packages/flutter/test/material/outline_button_test.dart [hunk 2]
```diff
@@ -44,78 +44,98 @@ void main() {
...
+  testWidgets('OutlineButton shape and border component overrides', (WidgetTester tester) async {
+    const Color fillColor = const Color(0xFF00FF00);
+    const Color borderColor = const Color(0xFFFF0000);
+    const Color highlightedBorderColor = const Color(0xFF0000FF);
+    const Color disabledBorderColor = const Color(0xFFFF00FF);
     const double borderWidth = 4.0;
 
-    await tester.pumpWidget(
-      new Directionality(
+    Widget buildFrame(Key key, VoidCallback onPressed) {
```
Comment (HansMuller) [inline packages/flutter/test/material/outline_button_test.dart]: The key parameter doesn't seem necessary, wouldn't `find.byType(OutlineButton)` be sufficient?

Also: even if this method only has one parameter, the calls will be a little easier to read with named parameters:
```
buildFrame(onPressed: null)
```

File: packages/flutter/test/material/outline_button_test.dart [hunk 3]
```diff
@@ -44,78 +44,98 @@ void main() {
...
+    // Pump a new widget with no onPressed to make it enabled.
+    const Key enabledKey = Key('enabled');
+    await tester.pumpWidget(
+      buildFrame(enabledKey, () { }),
+    );
+
+    final Finder enabledOutlineButton = find.byKey(enabledKey);
+    expect(tester.widget<OutlineButton>(enabledOutlineButton).enabled, true);
+
+    await tester.pump();
```
Comment (HansMuller) [inline packages/flutter/test/material/outline_button_test.dart]: Use `tester.pumpAndSettle()` when you want to wait until all pending animations (and etc) have finished.

**Review by HansMuller — COMMENTED** [review]
This looks good. Just some minor suggestions.

File: packages/flutter/test/material/outline_button_test.dart (hunk 1 above)
Comment (clocksmith) [reply packages/flutter/test/material/outline_button_test.dart]: clarified here that it is no-op just to make it enabled, (as opposed to null which makes it disabled)

File: packages/flutter/test/material/outline_button_test.dart (hunk 3 above)
Comment (clocksmith) [reply packages/flutter/test/material/outline_button_test.dart]: Done.

File: packages/flutter/test/material/outline_button_test.dart (hunk 2 above)
Comment (clocksmith) [reply packages/flutter/test/material/outline_button_test.dart]: key removed, changed to named params

**Comment by clocksmith** [comment]
Updated to respond to comments and add some additional comments to clarify flow.

The initial drawing of the button does not require a "pumpAndSettle" or pump with duration in order for it to be rendered, but pumping new widgets do require durations or settling to change render states. Why is this?

**Comment by HansMuller** [comment]
I generally use pumpAndSettle() when I'm not interested in any intermediate animation frames. It's not necessary when everything happens in one frame. Specifying a pump duration is useful when you want to sample a specific frame within an animation.

**Review by HansMuller — APPROVED** [review]
LGTM, but you'll need to fix up the analyzer errors.

**Pull request merged**